### PR TITLE
EventInvoice: Add seperate resourcelist class, docs, permissions for …

### LINKED ISSUE #4111: Event Invoice: Add seperate resourcelist class, docs, permissions for direct endpoints
Parent issue #4101.

### PR DESCRIPTION
**Pull request by enigmaeth**
Fixes #4111.

### CODE REVIEW

File: app/api/event_invoices.py [hunk 1]
```diff
@@ -120,13 +131,13 @@ def before_create_object(self, data, view_kwargs):
             data['discount_code_id'] = discount_code.id
 
     view_kwargs = True
-    decorators = (is_admin,)
```
Comment (SaptakS) [inline app/api/event_invoices.py]: Should event-invoices be visible to everyone? I think only organiser. @niranjan94 ?

Comment (poush) [inline app/api/event_invoices.py]: I think the list should be accessible only to server admins. However, the organizer should be able to see their own invoices. So the permission would be the organizer but in the before_get we have to ensure if it is direct route ( no event id present ) then user must be admin.

Comment (poush) [inline app/api/event_invoices.py]: But once @niranjan94 also told me the Event Invoices are managed by server admins and the generated invoices are sent to organizers by mail, so may be only server admins can see through APIs

File: app/api/event_invoices.py [hunk 2]
```diff
@@ -71,6 +74,21 @@ class Meta:
                                   type_='discount-code')
 
 
+class EventInvoiceListPost(ResourceList):
+    """
+    List and create Tracks
+    """
+    def before_post(self, args, kwargs, data):
+        require_relationship(['event'], data)
```
Comment (poush) [inline app/api/event_invoices.py]: I think user is also required as a relation, @SaptakS @shubham-padia please confirm

Comment (enigmaeth) [inline app/api/event_invoices.py]: @poush @SaptakS @shubham-padia 
I also think that user should be required. Plus updated GET permissions to `organizer` from [dev-handbook](https://github.com/fossasia/open-event/blob/master/dev-handbook/resources-permissions.md#event-invoices---eg-v1event-invoice). Please check.

**Comment by enigmaeth** [comment]
@poush @SaptakS Please review.

File: app/api/event_invoices.py [hunk 3]
```diff
@@ -71,6 +74,22 @@ class Meta:
...
+        require_relationship(['user'], data)
```
Comment (poush) [inline app/api/event_invoices.py]: It should be `require_relationship(['user', 'event'], data)`

File: docs/api/api_blueprint.apib [hunk 4]
```diff
@@ -2598,6 +2597,14 @@ Create a new Event Invoice using an event_id.
...
             {
               "data": {
+                "relationships":{
+                  "event":{
+                    "data":{
+                      "type":"event",
+                      "id":"1"
+                    }
+                  }
+                },
```
Comment (poush) [inline docs/api/api_blueprint.apib]: user relationship here

**Comment by enigmaeth** [comment]
@poush Resolved. Please check.

File: app/api/event_invoices.py (hunk 3 above)
Comment (enigmaeth) [inline app/api/event_invoices.py]: Resolved.

File: docs/api/api_blueprint.apib (hunk 4 above)
Comment (enigmaeth) [inline docs/api/api_blueprint.apib]: Resolved.

**Comment by enigmaeth** [comment]
@SaptakS Can you please review this now?

**Pull request merged**